powersimdata/input/transform_profile.py

```diff
@@ -1,111 +1,98 @@
-            the change table.
+        :return: (*pandas.DataFrame*) -- power output profile with additional columns
+            corresponding to new generators inserted to the grid via the change table.
         """
         new_plant_ids, neighbor_ids, scaling = [], [], []
         for i, entry in enumerate(self.ct["new_plant"]):
             if entry["type"] in self.scale_keys[resource]:
-                new_plant_id = self.grid.plant.index[-self.n_new_plant + i]
-                new_plant_ids.append(new_plant_id)
-                neighbor_id = entry["plant_id_neighbor"]
-                neighbor_ids.append(neighbor_id)
-                scaling.append(entry["Pmax"] / self.base_plant.loc[neighbor_id, "Pmax"])
-
-        if len(new_plant_ids) > 0:
-            neighbor_profiles = profile[neighbor_ids]
-            new_profiles = neighbor_profiles.multiply(scaling, axis=1)
-            new_profiles.columns = new_plant_ids
-            joined_profiles = profile.join(new_profiles)
-            return joined_profiles
-        else:
-            return profile
+                new_plant_ids.append(self.grid.plant.index[-self.n_new_plant + i])
+                neighbor_ids.append(entry["plant_id_neighbor"])
+                scaling.append(entry["Pmax"])
 
-    def _scale_plant_profile(self, profile, resource):
-        """Scales power output according to change table.
-
-        :param pandas.DataFrame profile: power output profile with plant
-            identification number as columns and UTC timestamp as index.
-        :param resource: fuel type.
-        :return: (*pandas.DataFrame*) -- scaled power output profile.
-        """
-        for r in self.scale_keys[resource]:
-            if r in self.ct.keys() and "zone_id" in self.ct[r].keys():
-                if self.n_new_plant > 0:
-                    type_in_zone = self.grid.plant[: -self.n_new_plant].groupby(
-                        ["zone_id", "type"]
-                    )
-                else:
-                    type_in_zone = self.grid.plant.groupby(["zone_id", "type"])
-                for z, f in self.ct[r]["zone_id"].items():
-                    plant_id = type_in_zone.get_group((z, r)).index.tolist()
-                    profile.loc[:, plant_id] *= f
-            if r in self.ct.keys() and "plant_id" in self.ct[r].keys():
-                for i, f in self.ct[r]["plant_id"].items():
-                    profile.loc[:, i] *= f
-
-        return profile
+        neighbor_profile = profile[neighbor_ids]
+        new_profile = neighbor_profile.multiply(scaling, axis=1)
+        new_profile.columns = new_plant_ids
+        return new_profile
 
     def _get_demand_profile(self):
-        """Returns scaled demand profile.
+        """Return scaled demand profile.
 
         :return: (*pandas.DataFrame*) -- data frame of demand.
         """
-        demand = self._input_data.get_data(self.scenario_info, "demand")
+        zone_id = interconnect2id[self.scenario_info["interconnect"]]
```

Comment:
Is this equivalent to `zone_id = sorted(self.grid.bus.zone_id.unique())`?

Reply:
Yes, indeed.

Reply:
Done